Add documentation on working with ScPCA portal data (#326)
AlexsLemonade/scpca-downstream-analyses: additional-docs/working-with-scpca-portal-data.md

# Working with ScPCA portal data

Downloads from the ScPCA Portal include gene expression data, a QC report, and associated metadata for each processed sample.
These files are delivered as a zip file.

> To find more information, visit the [ScPCA portal](https://scpca.alexslemonade.org/).

Reviewer: ```suggestion
To find more information on data available in the portal, visit the [ScPCA portal](https://scpca.alexslemonade.org/).
```

## Preparing the downloaded data for core workflow

> You can run the core downstream analysis workflow using the filtered RDS file and the metadata file found within the zip file downloaded from the ScPCA portal.

Reviewer: ```suggestion
The core downstream analysis workflow can take as input the `SingleCellExperiment` objects that are available in the filtered RDS files on the portal, files ending in `_filtered.rds`.
You will also need information obtained in the metadata file found within the zip file downloaded from the ScPCA portal, `single_cell_metadata.tsv`.
```
Let's refer to the actual file names here too just for added clarity and make sure to state what file is the input file.

Before running the workflow, you will need to create a project metadata file as a tab-separated value (TSV) file that contains the relevant data for your input files needed to run the workflow.
The metadata file should contain the following columns:

- `sample_id`, the unique ID for each piece of tissue or sample that cells were obtained from, which can be found in the `scpca_sample_id` column of the downloaded `single_cell_metadata.tsv` file.
- `library_id`, the unique ID used for each set of cells that has been prepped and sequenced separately, which can be found in the `scpca_library_id` columns of the downloaded `single_cell_metadata.tsv` file.

> - `filepath`, the full path to the `_filtered.rds` file containing the filtered `SingleCellExperiment` object.

Reviewer: ```suggestion
- `filepath`, the full path to the `_filtered.rds` file containing the filtered `SingleCellExperiment` object downloaded from the ScPCA portal.
```

Each library ID should have a unique `filepath`.

An example of the project metadata file would be as follows:

| sample_id | library_id | filepath |
| --------- | ---------- | --------- |
| SCPCS000122 | SCPCL000141 | example-data/SCPCS000122/SCPCL000141_filtered.rds |
| SCPCS000123	 | SCPCL000142 | example-data/SCPCS000123/SCPCL000142_filtered.rds |

### Running the workflow

The below code is an example of running the Snakemake workflow using the project-specific parameters.

```
snakemake --cores 2 \
  --use-conda \
  --config results_dir="<FULL PATH TO RESULTS DIRECTORY>" \
  project_metadata="<FULL PATH TO YOUR PROJECT METADATA TSV>" \

>   mito_file="<FULL PATH TO MITOCHONDRIAL GENES TXT FILE>"

Reviewer: ```suggestion
```
I think we could remove this right, because our reference file here is the same as what was used on the portal.

```

## Preparing the downloaded data to run additional modules

> You can run the downstream analysis clustering and genes of interest workflows using the processed RDS file and the metadata file found within the zip file downloaded from the ScPCA portal.

Reviewer: ```suggestion
You can run any of the additional modules available (e.g., clustering or the genes of interest workflows) using the processed RDS file (ending in `_processed.rds`) and the metadata file (`single_cell_metadata.tsv`) found within the zip file downloaded from the ScPCA portal.
```
Making this change so that when we add more modules we don't have to come back and edit this.

Before running any of these workflows, you will need to create a project metadata file as a tab-separated value (TSV) file that contains the relevant data for your input files needed to run the workflow.
The metadata file should contain the following columns:

- `sample_id`, the unique ID for each piece of tissue or sample that cells were obtained from, which can be found in the `scpca_sample_id` column of the downloaded `single_cell_metadata.tsv` file.
- `library_id`, the unique ID used for each set of cells that has been prepped and sequenced separately, which can be found in the `scpca_library_id` columns of the downloaded `single_cell_metadata.tsv` file.

> - `filepath`, the full path to the `_processed.rds` file containing the processed `SingleCellExperiment` object.

Reviewer: Technically we don't need this right? The `filepath` isn't required for these workflows as they use the `input_data_dir` and then expect the format of `sample_id/library_id_processed.rds` for the input files. I would remove this and then have a statement about the required `input_data_dir` and what the files should look like.

Each library ID should have a unique `filepath`.

An example of the project metadata file would be as follows:

| sample_id | library_id | filepath |
| --------- | ---------- | --------- |
| SCPCS000122 | SCPCL000141 | example-data/SCPCS000122/SCPCL000141_processed.rds |

> | SCPCS000123	 | SCPCL000142 | example-data/SCPCS000123/SCPCL000142_processed.rds |

Reviewer: Maybe instead of having a repeat of the table here, only include a table with two columns and then an example tree of the `input_data_dir` showing two samples downloaded from the portal and the file structure.

### Running the workflow

The below code is an example of running the Snakemake clustering workflow using the project-specific parameters.

```
snakemake --snakefile cluster.snakefile \ 
  --cores 2 \
  --use-conda \
  --config input_data_dir="<FULL PATH TO INPUT DATA DIRECTORY>" \
  results_dir="<FULL PATH TO RESULTS DIRECTORY>" \
  project_metadata="<FULL PATH TO YOUR PROJECT METADATA TSV>"
```

The below code is an example of running the Snakemake genes of interest workflow using the project-specific parameters.

```
snakemake --snakefile goi.snakefile \ 
  --cores 2 \
  --use-conda \
  --config input_data_dir="<FULL PATH TO INPUT DATA DIRECTORY>" \
  results_dir="<FULL PATH TO RESULTS DIRECTORY>" \
  project_metadata="<FULL PATH TO YOUR PROJECT METADATA TSV>"

> ```

Reviewer: I actually wonder now if we want to link to the `running the workflow` sections in the subsequent readme's instead of keeping it in two places.

# Working with ScPCA portal data

Downloads from the ScPCA Portal include gene expression data, a QC report, and associated metadata for each processed sample.

> These files are delivered as a zip file.

Reviewer: Maybe we should add a sentence here that states that each sample should include a file that can be used as input to the core workflow, or to skip directly to the modules, use the processed SCE object. And then include the file names. Just something right at the beginning to state which files go with which workflow.

## Preparing the downloaded data for core workflow

The core downstream analysis workflow can take as input the `SingleCellExperiment` objects that are available in the filtered RDS files on the portal, files ending in `_filtered.rds`.
You will also need information obtained in the metadata file found within the zip file downloaded from the ScPCA portal, `single_cell_metadata.tsv`.
Before running the workflow, you will need to create a project metadata file as a tab-separated value (TSV) file that contains the relevant data for your input files needed to run the workflow.
The metadata file should contain the following columns:

- `sample_id`, the unique ID for each piece of tissue or sample that cells were obtained from, which can be found in the `scpca_sample_id` column of the downloaded `single_cell_metadata.tsv` file.
- `library_id`, the unique ID used for each set of cells that has been prepped and sequenced separately, which can be found in the `scpca_library_id` columns of the downloaded `single_cell_metadata.tsv` file.
- `filepath`, the full path to the `_filtered.rds` file containing the filtered `SingleCellExperiment` object downloaded from the ScPCA portal.

> Each library ID should have a unique `filepath`.

Reviewer: ```suggestion
Each library ID should have a unique `filepath` and will probably look something like this for each file: `<full path to directory storing data downloaded from ScPCA/scpca_sample_id/scpca_library_id_filtered.rds`
```
Maybe provide an example of how to put together the path?

```

## Preparing the downloaded data to run additional modules

You can run any of the additional modules available (e.g., clustering or the genes of interest workflows) using the processed RDS file (ending in `_processed.rds`) and the metadata file (`single_cell_metadata.tsv`) found within the zip file downloaded from the ScPCA portal.
Before running any of these workflows, you will need to create a project metadata file as a tab-separated value (TSV) file that contains the relevant data for your input files needed to run the workflow.
The metadata file should contain the following columns:

- `sample_id`, the unique ID for each piece of tissue or sample that cells were obtained from, which can be found in the `scpca_sample_id` column of the downloaded `single_cell_metadata.tsv` file.
- `library_id`, the unique ID used for each set of cells that has been prepped and sequenced separately, which can be found in the `scpca_library_id` columns of the downloaded `single_cell_metadata.tsv` file.

An example of the project metadata file would be as follows:

| sample_id | library_id |
| --------- | ---------- |
| SCPCS000122 | SCPCL000141 |
| SCPCS000123	 | SCPCL000142 |

> The required `input_data_dir` should also contain the downloaded input files as follows:

Reviewer: Add a sentence before this to state what the `input_data_dir` is? Explain that it's a parameter that needs to be specified to run the workflow.

```
input_data_dir
   ├── SCPCS000122
   │   ├── SCPCL000141_processed.rds
   │   └── single_cell_metadata.tsv
   ├── SCPCS000123
   │   ├── SCPCL000142_processed.rds
   │   └── single_cell_metadata.tsv
```

### Running the workflow

> The below code is an example of running the additional workflow(s) using the project-specific parameters, where `<module.snakefile>` would be replaced with the name of the snakefile associated with the desired module.

Reviewer: I like this change!

# Working with ScPCA portal data

Downloads from the ScPCA Portal include gene expression data, a QC report, and associated metadata for each processed sample.

> These files are delivered as a zip file.

Reviewer: ```suggestion
```
I think you can remove this line.
